### PR TITLE
Resolve the vendor/wordpress-autoload.php file on Windows, add PHP 8.1-8.2 support

### LINKED ISSUE #28: Autoloader infrequently not included leading to class not found
### Description of the bug

Sometimes when loading `vendor/autoload.php`, the `vendor/wordpress-autoload.php` file is not properly loaded leading to the WordPress autoloader not being registered. A workaround for this is to require `vendor/wordpress-autoload.php` instead of `vendor/autoload.php` but this isn't as ideal.

### Steps To Reproduce

1. Require `vendor/autoload.php`.
2. Attempt to use a WordPress-autoloaded file.
3. Recieve a class not found error.
4. Switch to requiring `vendor/wordpress-autoload.php`.
5. See the class loading fine.

### Additional Information

This happens infrequently and usually on larger code bases with multiple uses of `composer-wordpress-autoloader` in it (for example, in the root of a project and 2+ plugins).

### PR DESCRIPTION
**Pull request by srtfisher**
## Changed

- Dropped PHP 7.4 support, adding PHP 8.2 and 3 support explicitly with testing.
- Added testing against Windows.

## Fixed

- Fixed issue with `wordpress-autoload.php` file not properly loading on Windows.

Fixes #28

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
 
 jobs:
   test:
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: true
       matrix:
-        os: [ubuntu-latest]
-        php: [7.4, 8.0, 8.1]
+        os: [ubuntu-latest, windows-2019]
```
Comment (kevinfodness) [inline .github/workflows/tests.yml:15]: 😮

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -1,15 +1,19 @@
 name: Tests
 
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - main
```
Comment (kevinfodness) [inline .github/workflows/tests.yml:6]: This will also trigger after the PR is merged. Is that intentional?

File: src/autoload.php [hunk 3]
```diff
@@ -7,7 +7,17 @@
  */
 
 $autoloadFiles = [
+    // Attempt to translate the /vendor/ in the directory path to the vendor directory.
     preg_replace('#/vendor/.*$#', '/vendor/wordpress-autoload.php', __DIR__),
+    preg_replace('/\\\vendor\\\.*$/', '/vendor/wordpress-autoload.php', __DIR__),
```
Comment (kevinfodness) [inline src/autoload.php:12]: This looks like one too many backslashes - shouldn't it be like this?
```suggestion
    preg_replace('/\\vendor\\.*$/', '/vendor/wordpress-autoload.php', __DIR__),
```

**Review by kevinfodness — APPROVED** [review]
A few questions, no blockers - thanks for the quick fix! 🍣

File: .github/workflows/tests.yml (hunk 2 above)
Comment (srtfisher) [reply .github/workflows/tests.yml:6]: Yes, I prefer to always have a status check on the `main` branch to get a valid status badge.

File: src/autoload.php (hunk 3 above)
Comment (srtfisher) [reply src/autoload.php:12]: I'm not a regex pro to know but in testing with `C:\Users\Sean Fisher\Local Sites\wordpress\app\public\wp-content\vendor\alleyinteractive\composer-wordpress-autoloader\src` as the `__DIR__`:

four slashes: `preg_replace('/\\\vendor\\\.*$/', '/vendor/wordpress-autoload.php', __DIR__),` -> `"C:\Users\Sean Fisher\Local Sites\wordpress\app\public\wp-content/vendor/wordpress-autoload.php`

two slashes: `preg_replace('/\\vendor\\\.*$/', '/vendor/wordpress-autoload.php', __DIR__),` -> `C:\Users\Sean Fisher\Local Sites\wordpress\app\public\wp-content\vendor\alleyinteractive\composer-wordpress-autoloader\src`

https://onlinephp.io?s=lc8_C8IwEAXw2UK_QwYhKtbu_sFBcHIrbgdyJocGYnNc0ha_vRHRRRfXx-P3eOstX7ksxtaJ2ii9W8IxkkRoCFu1d_FKAodg0KvGJYowBLEsFCMgM3B39s7AwJUJbaI2QU-tDQLoPd1djgRNcj2BCTcOWa4-QIVdCj6gzQtRjF6VRVn0KCfb3XhSFqPcupyE2KOhia4B3jjAYjau9Vzp-pXU3-gi38qN56_p_Bf2tv6mpqsH&v=8.2.13

/shrug

**Pull request merged**